gmso/external/convert_dataframe.py

```diff
@@ -0,0 +1,552 @@
+    `Atom0` to `newname` using a dictionary.
+        >>> df.rename(columns = {'Atom0':'newname'})
+    See https://pandas.pydata.org/pandas-docs/stable/getting_started/intro_tutorials/index.html for further information.
+    """
+    if columns is None:
+        columns = []
+    if not topology.is_typed():
+        raise GMSOError(
+            "This topology is not typed, please type this object before converting to a pandas dataframe"
+        )
+    outDict = {}  # dictionary of dataframes to write out
+
+    # Get columns from format methods
+    columnsDict = {}
+    connectionsList = [
+        "bonds",
+        "angles",
+        "dihedrals",
+        "impropers",
+    ]  # these can be handled generally
+    remove_duplicatesBool = False  # flag to remove duplicate parameters from the dataframe and put indices into a new column
+    if format == "default":
+        columnsDict = {
+            param: [
+                "name",
+                f"{param[:-1]}_type.member_classes",
+                f"{param[:-1]}_type.parameters",
+            ]
+            for param in connectionsList
+        }
+        columnsDict["sites"] = [
+            "name",
+            "atom_type.name",
+            "atom_type.parameters",
+            "charge",
+            "mass",
+        ]
+        if isinstance(columns, list):
+            columnsDict = {
+                key: columnsDict[key] + columns for key in columnsDict.keys()
+            }  # add in any provided columns
+    elif format == "specific_columns":
+        assert parameter != "all", (
+            f"When formatting for specific columns, please set parameter argument to be one of {['sites']+connectionsList}."
+            "Otherwise use a format of default."
+        )
+        columnsDict = {parameter: columns}
+    elif format == "publication":
+        columnsDict = {
+            param: [
+                "name",
+                f"{param[:-1]}_type.member_classes",
+                f"{param[:-1]}_type.parameters",
+            ]
+            for param in connectionsList
+        }
+        columnsDict["sites"] = [
+            "name",
+            "atom_type.name",
+            "atom_type.parameters",
+            "charge",
+            "mass",
+        ]
+        remove_duplicatesBool = True
+    elif format == "remove_duplicates":
+        assert parameter != "all", (
+            f"When formatting for specific columns, please set parameter argument to be one of {['sites']+connectionsList}."
+            "Otherwise use a format of default."
+        )
+        if not columns and parameter == "sites":  # default values
+            columns = ["atom_type.name"]
+        elif not columns and parameter in connectionsList:
+            columns = [f"{parameter[:-1]}_type.member_classes"]
+        columnsDict = {parameter: columns}
+    else:
+        raise ValueError(
+            f"Please provide formt=['default', 'specific_columns', 'publication']"
```

Comment:
Small typo here, I think this should say format

Reply:
Also, we should add `remove_duplicates` as an option here correct? I think this error message could be a little more clear. Something like `"Available options for format are 'default', 'specific_columns', 'publication', or `remove_duplicates`."` 

Also, the `f` string isn't doing anything here, do we want to print out what they passed to `format`?

Reply:
Agreed